futures-util/src/try_stream/mod.rs

```diff
@@ -15,6 +25,97 @@ impl<S: TryStream> TryStreamExt for S {}
 
 /// Adapters specific to `Result`-returning streams
 pub trait TryStreamExt: TryStream {
+    /// Wraps the current future in a new future which converts the error type
+    /// into the one provided.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use futures::executor::block_on;
+    /// use futures::prelude::*;
+    ///
+    /// let mut stream =
+    ///     stream::iter(vec![Ok(()), Err(5i32)])
+    ///         .err_into::<i64>();
+    ///
+    /// assert_eq!(block_on(stream.try_next()), Ok(Some(())));
+    /// assert_eq!(block_on(stream.try_next()), Err(5i64));
+    /// ```
+    fn err_into<E>(self) -> ErrInto<Self, E>
+    where
+        Self: Sized,
+        Self::Error: Into<E>
+    {
+        ErrInto::new(self)
+    }
+
+    /// Creates a future that attempts to resolve the next item in the stream.
+    /// If an error is encountered before the next item, the error is returned
+    /// instead.
+    ///
+    /// This is similar to the `Stream::next` combinator, but returns a
+    /// `Result<Option<T>, E>` rather than an `Option<Result<T, E>>`, making
+    /// for easy use with `?`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use futures::executor::block_on;
+    /// use futures::prelude::*;
+    ///
+    /// let mut stream = stream::iter(vec![Ok(()), Err(())]);
+    ///
+    /// assert_eq!(block_on(stream.try_next()), Ok(Some(())));
+    /// assert_eq!(block_on(stream.try_next()), Err(()));
+    /// ```
+    fn try_next(&mut self) -> TryNext<'_, Self>
+        where Self: Sized + Unpin,
+    {
+        TryNext::new(self)
+    }
+
+    /// Attempts to run this stream to completion, executing the provided
+    /// asynchronous closure for each element on the stream.
+    ///
+    /// The closure provided will be called for each item this stream produces,
+    /// yielding a future. That future will then be executed to completion
+    /// before moving on to the next item.
+    ///
+    /// The returned value is a `Future` where the `Output` type is
+    /// `Result<(), Self::Error>`. If any of the intermediate futures or the
+    /// stream return an error, this future will return immediately with an
+    /// error.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use futures::executor::block_on;
+    /// use futures::prelude::*;
+    ///
+    /// let mut x = 0;
+    ///
+    /// {
+    ///     let fut = stream::repeat(Ok(1)).try_for_each(|item| {
+    ///         x += item;
+    ///         if x == 3 {
+    ///             future::ready(Err(()))
+    ///         } else {
+    ///             future::ready(Ok(()))
```

Comment:
Maybe
```rust
if x == 3 { return future::ready(Err(())); }
future::ready(Ok(()))
```

Reply:
changed to `future::ready(if x == 3 { Err(()) } else { Ok(()) })`